django/test/utils.py

```diff
@@ -841,6 +845,37 @@ def disable(self):
         setattr(Options, 'default_apps', self.old_apps)
 
 
+class TimeKeeper:
+    def __init__(self):
+        self.records = {}
+
+    @contextmanager
+    def timed(self, name):
+        start_time = time.perf_counter()
+        if name not in self.records.keys():
+            self.records[name] = []
+        try:
+            yield
+        finally:
+            end_time = time.perf_counter() - start_time
+            self.records[name].append(end_time)
+
+    def results(self):
```

Comment:
Should we name this `print_results` or move the `sys.stderr.write()` call outside?
(I don't know if there would be a use for returning the generated text for something other than printing it out.)